observation_portal/observations/viewsets.py

```diff
@@ -82,10 +82,6 @@ def perform_create(self, serializer):
             using its observation id as the downtime reason
         """
         observation = serializer.save(submitter=self.request.user, submitter_id=self.request.user.id)
-        # Debit the realtime time allocation hours
```

Comment:
Would wrapping this whole thing in a database transaction help make sure there are no edge-cases with the clean-up?

Reply:
I think that would work, but I'm worried about having an external API call within a transaction since that seems like bad practice since the external api call could hang or talk a while and block other db writes...